test_conformance/geometrics/test_geometrics_fp.cpp

```diff
@@ -0,0 +1,2240 @@
+//
```

Comment:
This new file is pretty monolithic, although it does solve the previous code duplication.

Is there any way to break it down into separate files to make it the code easier to navigate? The template instantiation
might be problematic. Thinking splitting out common/helper code, or verification code, or setup code.

Reply:
Absolutely. I see four logical parts of this test and my proposition is to create separate source files:

1) Cross test
2) Dot / FastDistance / Distance tests
3) Length / FastLength
4) Normalize / FastNormalize

... plus header file for base structures. As visible, we could divide even more if you think that could impact readability of the code.

Reply:
That sounds like a good organization to me, I don't think we need to go more granular than your proposal.